File: src/main/java/hdzi/editstarters/ui/InitializrUrlDialog.java
package hdzi.editstarters.ui;

import com.intellij.openapi.ui.InputValidator;
import com.intellij.openapi.ui.Messages;

@Deprecated
class InitializrUrlDialog {
    private final InputValidator urlVerifier = new InputValidator() {
        @Override
        public boolean checkInput(String inputString) {
            return inputString.matches("^https?://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]");
        }

        @Override
        public boolean canClose(String inputString) {
            return true;
        }
    };

    public String getUrl() {
        return Messages.showInputDialog(
                null,
                "Spring Initializr Url",
                null,
                "https://start.spring.io/",
                urlVerifier
        );
    }
}

File: src/main/kotlin/hdzi/editstarters/ui/InitializrUrlDialog.kt
package hdzi.editstarters.ui

import com.intellij.openapi.ui.InputValidator
import com.intellij.openapi.ui.Messages

class InitializrUrlDialog {
    private val urlVerifier = object : InputValidator {
        private val URL_PATTERN = "^https?://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]".toRegex()

        override fun checkInput(inputString: String): Boolean = inputString.matches(this.URL_PATTERN)

        override fun canClose(inputString: String) = true
    }

    private val defaultUrl = "https://start.spring.io/"

    val url: String
        get() = Messages.showInputDialog(
            null,
            "Spring Initializr Url",
            null,
            defaultUrl,
            urlVerifier
        ) ?: defaultUrl
}
